cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/concepts/Operation.kt

```diff
@@ -49,5 +49,15 @@ abstract class Operation(
         return other is Operation && super.equals(other) && other.concept == this.concept
     }
 
+    override fun equalWithoutUnderlying(other: OverlayNode): Boolean {
```

Comment:
I think we can exclude the underlying node in the default equals instead of having this very very weird piece of code.

Reply:
Not sure if that's a good idea because if there is an `underlyingNode` and it's different, the overlay node should also be different

Reply:
What we could/should do instead is to implement this weird logic and just call it in `equals` together with the check for the `underlyingNode`